src/server/scripts/World/go_scripts.cpp

```diff
@@ -1569,6 +1570,124 @@ class go_pirate_day_music : public GameObjectScript
     }
 };
 
+/*####
+## go_bells
+####*/
+
+enum BellHourlySoundFX
+{
+    BELLTOLLHORDE = 6595, // Horde
+    BELLTOLLTRIBAL = 6675,
+    BELLTOLLALLIANCE = 6594, // Alliance
+    BELLTOLLNIGHTELF = 6674,
+    BELLTOLLDWARFGNOME = 7234,
+    BELLTOLLKHARAZHAN = 9154 // Kharazhan
+};
+
+enum BellHourlySoundAreas
+{
+    B_UNDERCITY_AREA = 1497,
+    B_IRONFORGE_1_AREA = 809,
+    B_IRONFORGE_2_AREA = 1,
+    B_DARNASSUS_AREA = 1657,
+    B_TELDRASSIL_ZONE = 141,
+    B_KHARAZHAN_MAPID = 532
+};
+
+enum BellHourlyMisc
+{
+    GO_HORDE_BELL = 175885,
+    GO_ALLIANCE_BELL = 176573,
+    GO_KHARAZHAN_BELL = 182064
+};
+
+enum BellHourlyEvent
+{
+    EVENT_RING_BELL = 1
+};
+
+class go_bells : public GameObjectScript
+{
+public:
+    go_bells() : GameObjectScript("go_bells") { }
+
+    struct go_bellsAI : public GameObjectAI
+    {
+        go_bellsAI(GameObject* go) : GameObjectAI(go)
+        {
+            _started = false;
+            _rings = 0;
+        };
+
+        void UpdateAI(uint32 diff) override
+        {
+            _events.Update(diff);
+
+            time_t time = sWorld->GetGameTime();
+            tm localTm;
+            localtime_r(&time, &localTm);
+
+            if ((localTm.tm_min == 0 && localTm.tm_sec == 0) && !_started)
+            {
+                _rings = (localTm.tm_hour - 1) % 12 + 1;
+                _events.ScheduleEvent(EVENT_RING_BELL, Seconds(1));
+                _started = true;
+            }
+
+            while (uint32 eventId = _events.ExecuteEvent())
+            {
+                switch (eventId)
+                {
+                case EVENT_RING_BELL:
+                {
+                    if (go->GetEntry() == GO_HORDE_BELL)
```

Comment:
and now you've overdone it, the one liners doing go->PlayDirectSound dont need them (only the outer ones do)